code/modules/mob/living/basic/snails/snail.dm

```diff
@@ -0,0 +1,113 @@
+#define SNAIL_MOVEMENT_DISTANCE 5
+#define SNAIL_MOVEMENT_TIME 10 SECONDS
+
+/mob/living/basic/snail
+	name = "snail"
+	desc = "Is petting this thing sanitary?"
+	icon_state = "snail"
+	icon_living = "snail"
+	icon_dead = "snail"
+	base_icon_state = "snail"
+	held_state = "snail"
+	basic_mob_flags = FLIP_ON_DEATH
+	head_icon = 'icons/mob/clothing/head/pets_head.dmi'
+	icon = 'icons/mob/simple/pets.dmi'
+	butcher_results = list(/obj/item/food/meat/rawcutlet = 1)
+	mob_biotypes = MOB_ORGANIC
+	health = 30
+	maxHealth = 30
+	speed = 6
+	verb_say = "gurgles"
+	verb_ask = "gurgles curiously"
+	can_be_held = TRUE
+	verb_exclaim = "gurgles loudly"
+	verb_yell = "gurgles loudly"
+	worn_slot_flags = ITEM_SLOT_HEAD
+	faction = list(FACTION_NEUTRAL, FACTION_MAINT_CREATURES)
+	ai_controller = /datum/ai_controller/basic_controller/snail
+
+/mob/living/basic/snail/Initialize(mapload)
+	. = ..()
+	var/static/list/loc_connections = list(COMSIG_ATOM_ENTERED = PROC_REF(on_entered))
+	AddElement(/datum/element/connect_loc, loc_connections)
+	var/static/list/eatable_food = list(
+		/obj/item/food/grown,
+		/obj/item/food/appleslice,
+	)
+
+	var/static/list/innate_actions = list(
+		/datum/action/cooldown/mob_cooldown/shell_retreat = BB_SNAIL_RETREAT_ABILITY,
+	)
+
+	grant_actions_by_list(innate_actions)
+	AddElement(/datum/element/ai_retaliate)
+	ai_controller.set_blackboard_key(BB_BASIC_FOODS, typecacheof(eatable_food))
+	AddElement(/datum/element/basic_eating, food_types = eatable_food)
+	create_reagents(100, REAGENT_HOLDER_ALIVE)
+	RegisterSignal(reagents, COMSIG_REAGENTS_HOLDER_UPDATED, PROC_REF(on_reagents_update))
+
+/mob/living/basic/snail/proc/on_entered(datum/source, obj/effect/decal/cleanable/food/salt/potential_salt)
+	SIGNAL_HANDLER
+	if(istype(potential_salt))
+		death() //immediately perish
```

Comment:
Maybe print a visible message here (after checking if it's already dead of course, technically you keep calling `death()` on dead snails that are dragged through salt too)